crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -555,10 +568,29 @@ where
             } else {
                 state.head_block_hash
             };
-            self.sync.set_pipeline_sync_target(target);
+
+            // we need to first check the buffer for the head and its ancestors
+            let lowest_unknown_hash = self
+                .blockchain
+                .lowest_buffered_ancestor(target)
+                .map(|block| block.parent_hash)
+                .unwrap_or(target);
+
+            trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering pipeline with target instead of downloading");
+
+            self.sync.set_pipeline_sync_target(lowest_unknown_hash);
         } else {
+            // we need to first check the buffer for the head and its ancestors
+            let lowest_unknown_hash = self
+                .blockchain
+                .lowest_buffered_ancestor(state.head_block_hash)
+                .map(|block| block.parent_hash)
+                .unwrap_or(state.head_block_hash);
```

Comment:
same pattern as above, right? so function def useful

Reply:
yep, just added `lowest_buffered_ancestor_or` and replaced all uses